lib/run_loop/directory.rb

```diff
@@ -52,19 +52,29 @@ def self.directory_digest(path, options={})
         raise ArgumentError, "Expected '#{path}' to be a directory"
       end
 
-      entries = self.recursive_glob_for_entries(path)
+      entries = self.recursive_glob_for_entries(path).sort
 
       if entries.empty?
         raise ArgumentError, "Expected a non-empty dir at '#{path}' found '#{entries}'"
       end
 
       debug = RunLoop::Environment.debug?
 
-      sha = OpenSSL::Digest::SHA256.new
-      entries.each do |file|
-        unless self.skip_file?(file, 'SHA1', debug)
+      file_shas = []
+      cumulative = OpenSSL::Digest::SHA256.new
```

Comment:
is cumulative here to stay? in that case I think we should document it being returned by the function.